modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/SparkUtils.scala

```diff
@@ -59,44 +58,48 @@ private[processing] object SparkUtils {
     writer.sparkConfig ++ config.conf + (gcpUserAgentKey -> gcpUserAgentValue)
   }
 
-  def saveDataFrameToDisk[F[_]: Sync](
+  def initializeLocalDataFrame[F[_]: Sync](spark: SparkSession, viewName: String): F[Unit] =
+    for {
+      _ <- Logger[F].debug(s"Initializing local DataFrame with name $viewName")
+      _ <- Sync[F].blocking {
+             spark.emptyDataFrame.createTempView(viewName)
+           }
+    } yield ()
+
+  def localAppendRows[F[_]: Sync](
     spark: SparkSession,
+    viewName: String,
     rows: NonEmptyList[Row],
     schema: StructType
-  ): F[DataFrameOnDisk] = {
-    val count = rows.size
+  ): F[Unit] =
     for {
-      viewName <- Sync[F].delay("v" + UUID.randomUUID.toString.replaceAll("-", ""))
-      _ <- Logger[F].debug(s"Saving batch of $count events to local disk")
+      _ <- Logger[F].debug(s"Saving batch of ${rows.size} events to local DataFrame $viewName")
       _ <- Sync[F].blocking {
              spark
                .createDataFrame(rows.toList.asJava, schema)
                .coalesce(1)
```

Comment:
This `coalesce` is still 1 and it's ok, right?

Reply:
Yes here it is ok.  It means the individual small dataframes, which don't need to be split any further.